src/busctl/busctl.c

```diff
@@ -1361,6 +1362,12 @@ static int monitor(int argc, char **argv, int (*dump)(sd_bus_message *m, FILE *f
                         dump(m, stdout);
                         fflush(stdout);
 
+                        if (arg_num_matches != UINT64_MAX && --arg_num_matches == 0) {
```

Comment:
i hate side-effects of if checks. please separate the subtraction into its own line. this is really confusing otherwise.